src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3497,29 +3497,31 @@ private BoundNode BindConstructorBody(ConstructorDeclarationSyntax constructor,
 
             bool thisInitializer = initializer?.IsKind(SyntaxKind.ThisConstructorInitializer) == true;
             if (!thisInitializer &&
-                hasAnyRecordConstructors())
+                hasRecordPrimaryConstructor())
             {
-                var constructorSymbol = (MethodSymbol)this.ContainingMember();
-                if (!constructorSymbol.IsStatic &&
+                if (isInstanceConstructor(out MethodSymbol constructorSymbol) &&
```

Comment:
nit: merge into outer `if` now that we can?

Reply:
Let's address separately if necessary.